include/tudat/astro/system_models/geometryUtils.h

```diff
@@ -0,0 +1,201 @@
+/*    Copyright (c) 2010-2019, Delft University of Technology
+*    All rigths reserved
+*
+*    This file is part of the Tudat. Redistribution and use in source and
+*    binary forms, with or without modification, are permitted exclusively
+*    under the terms of the Modified BSD license. You should have received
+*    a copy of the license with this file. If not, please or visit:
+*    http://tudat.tudelft.nl/LICENSE.
+*/
+#ifndef TUDAT_GEOMETRYUTILS_H
+#define TUDAT_GEOMETRYUTILS_H
+
+#include <map>
+#include <iostream>
+#include <memory>
+
+namespace tudat
+{
+
+namespace system_models
+{
+class Triangle2d
+{
+private:
+    Eigen::Vector2d vertexA_, vertexB_, vertexC_;
+
+public:
+   // default constructor
+   Triangle2d( ):
+   vertexA_( Eigen::Vector2d::Zero() ),
+   vertexB_( Eigen::Vector2d::Zero() ),
+   vertexC_( Eigen::Vector2d::Zero() )
+   { }
+
+   Triangle2d( const Eigen::Vector2d& vertexA,
+               const Eigen::Vector2d& vertexB,
+               const Eigen::Vector2d& vertexC ): 
+               vertexA_( vertexA ), 
+               vertexB_( vertexB ), 
+               vertexC_( vertexC ) 
+   { }
+
+   Eigen::Vector2d getVertexA( ) const 
+   {
+       return vertexA_;
+   }
+   Eigen::Vector2d getVertexB( ) const
+   {
+       return vertexB_;
+   }
+   Eigen::Vector2d getVertexC( ) const
+   {
+       return vertexC_;
+   }
+};
+
+class Triangle3d
+{
+private:
+    Eigen::Vector3d vertexA_, vertexB_, vertexC_;
+    Eigen::Vector3d versorL_, versorM_;
+
+public:
+   Triangle3d( ) :
+   vertexA_( Eigen::Vector3d::Zero( ) ),
+   vertexB_( Eigen::Vector3d::Zero( ) ),
+   vertexC_( Eigen::Vector3d::Zero( ) ),
+   versorL_( Eigen::Vector3d::Zero( ) ),
+   versorM_( Eigen::Vector3d::Zero( ) )
+   { }
+   
+   Triangle3d( const Eigen::Vector3d& vertexA,
+               const Eigen::Vector3d& vertexB,
+               const Eigen::Vector3d& vertexC ): 
+               vertexA_( vertexA ),
+               vertexB_( vertexB ),
+               vertexC_( vertexC ) 
+   {  
+        // compute unit vectors l, m
+        versorL_ = ( vertexB_ - vertexA_ ).normalized( ) ;
+        Eigen::Vector3d edgeAB = vertexB_ - vertexA_;
+        Eigen::Vector3d edgeAC = vertexC_ - vertexA_;
+        Eigen::Vector3d normal = edgeAB.cross(edgeAC).normalized( );
+        versorM_ = normal.cross( versorL_ ).normalized( ); 
+    }
+
+   Eigen::Vector3d getVertexA( ) const
+   {
+       return vertexA_;
+   }
+   Eigen::Vector3d getVertexB( ) const
+   {
+       return vertexB_;
+   }
+   Eigen::Vector3d getVertexC( ) const
+   {
+       return vertexC_;
+   }
+   Eigen::Vector3d getVersorL( ) const
+   {
+       return versorL_;
+   }
+   Eigen::Vector3d getVersorM( ) const
+   {
+       return versorM_;
+   }
+};
+
+class ParallelProjection
+{
+private:
+    double minCoordinateL_, maxCoordinateL_, minCoordinateM_, maxCoordinateM_;

```

Comment:
min-> minimum
max-> maximum

Do 'L' and 'M' and 'lambda' have a physical/geometrical meaning that can be used for the naming?